### ISSUE TITLE
UUID variable ${uuidType} not used in change set sample data

### ISSUE THREAD

**Issue by harald2013** [jhipster/generator-jhipster#10855]
##### **Overview of the issue**

When generating code with a JDL file and using the UUID type, the create created code in the change set file config/liquibase/changelog/20191130021938_added_entity_Item.xml uses
<column name="identifier" type="${uuidType}"> for createTable but <column name="identifier" type="string"/> for loadData.

When starting the application, an error occurs when liquibase tries to import the data:
2019-11-30 17:08:42.278 ERROR 27316 --- [monolith-task-1] i.g.j.c.liquibase.AsyncSpringLiquibase   : Liquibase could not start correctly, your database is NOT ready: Migration failed for change set config/liquibase/changelog/20191130021938_added_entity_Item.xml::20191130021938-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.item(id, identifier, version, type, group_key, key, sort_order, text_1, text_2, text_3, obligation) VALUES(1, '5cf2bc99-2721-407d-992b-a00fbdf302f2', 61748, 'Home Loan Account', 'web-readiness', 'Namibia', 39986, 'Gorgeous Frozen Pants', 'JHipster is a development platform to generate, develop and deploy Spring Boot + Angular / React / Vue Web applications and Spring microservices.', 'JHipster is a development platform to generate, develop and deploy Spring Boot + Angular / React / Vue Web applications and Spring microservices.', 'Common') was aborted: FEHLER: Spalte ╗identifier½ hat Typ uuid, aber der Ausdruck hat Typ character varying
  Hinweis: Sie m³ssen den Ausdruck umschreiben oder eine Typumwandlung vornehmen.

It's german, but it says that identifier is uuid but the expression is character varying. You must do a value conversion or change the expression. 

When I change the type="String to "type="${uuidType}", everything works fine.

I'm also using the type TextBlob in my JDL. For createTable type="${clobType}" is generated for loadData type="clob". This akctually works, but maybe this should also be equal?

##### **Motivation for or Use Case**

Every time when i change my entities an recreate them, i have to fix the change set files.

##### **Reproduce the error**

generate and entity with an UUID und user test data, start the application and the error occurs on startup.

##### **Related issues**

found nothing related

##### **Suggest a Fix**

Use "type="${uuidType}" also in loadData, check if type="clob" should also be repaced by type="${clobType}" 

##### **JHipster Version(s)**

JHipster v6.5.0

##### **JHipster configuration**

INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
cs-prototype-monolith@0.0.1-SNAPSHOT C:\Users\haral\Code\compliance-suite\cs-prototype-monolith
`-- generator-jhipster@6.5.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.rh.cs"
    },
    "jhipsterVersion": "6.5.0",
    "applicationType": "monolith",
    "baseName": "CsPrototypeMonolith",
    "packageName": "com.rh.cs",
    "packageFolder": "com/rh/cs",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false,
    "blueprints": []
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Structure {
  identifier UUID required unique,
  version Integer required,
  titleCols String required minlength(3) maxlength(30),
  titleFull String required minlength(3) maxlength(300),
  textLong TextBlob,
  sortOrder Integer
}
entity Line {
  identifier UUID required unique,
  version Integer required,
  sortOrder Integer required,
  level Integer required
}
entity Item {
  identifier UUID required unique,
  version Integer required,
  type String required minlength(3) maxlength(50),
  groupKey String required minlength(3) maxlength(50),
  key String required maxlength(30),
  sortOrder Integer required,
  text1 String maxlength(1000),
  text2 TextBlob,
  text3 TextBlob,
  obligation String maxlength(30)
}
relationship OneToMany {
  Structure{line} to Line{structure(titleCols) required}
}
relationship ManyToOne {
  Line{item(key) required} to Item
}

paginate Structure, Line with pagination
service Structure, Line, Item with serviceClass
filter Structure, Line, Item

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_231"
Java(TM) SE Runtime Environment (build 1.8.0_231-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.231-b11, mixed mode)

git version 2.19.1.windows.1

node: v8.12.0

npm: 6.4.1

yeoman: 3.1.0

Docker version 19.03.5, build 633a0ea

docker-compose version 1.24.1, build 4667896b

INFO! Congratulations, JHipster execution is complete!

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

{
    "name": "Structure",
    "fields": [
        {
            "fieldName": "identifier",
            "fieldType": "UUID",
            "fieldValidateRules": [
                "required",
                "unique"
            ]
        },
        {
            "fieldName": "version",
            "fieldType": "Integer",
            "fieldValidateRules": [
                "required"
            ]
        },
        {
            "fieldName": "titleCols",
            "fieldType": "String",
            "fieldValidateRules": [
                "required",
                "minlength",
                "maxlength"
            ],
            "fieldValidateRulesMinlength": 3,
            "fieldValidateRulesMaxlength": 30
        },
        {
            "fieldName": "titleFull",
            "fieldType": "String",
            "fieldValidateRules": [
                "required",
                "minlength",
                "maxlength"
            ],
            "fieldValidateRulesMinlength": 3,
            "fieldValidateRulesMaxlength": 300
        },
        {
            "fieldName": "textLong",
            "fieldType": "byte[]",
            "fieldTypeBlobContent": "text"
        },
        {
            "fieldName": "sortOrder",
            "fieldType": "Integer"
        }
    ],
    "relationships": [
        {
            "relationshipType": "one-to-many",
            "otherEntityName": "line",
            "otherEntityRelationshipName": "structure",
            "relationshipName": "line"
        }
    ],
    "changelogDate": "20191130021936",
    "entityTableName": "structure",
    "dto": "no",
    "pagination": "pagination",
    "service": "serviceClass",
    "jpaMetamodelFiltering": true,
    "fluentMethods": true,
    "readOnly": false,
    "clientRootFolder": "",
    "applications": "*"
}
##### **Browsers and Operating System**

not relevant

-   [x ] Checking this box is mandatory (this is just to show you read everything)

**Comment by ruddell** [comment]
Thank you for reporting and filling out the issue template.  This looks to be fixed in the latest release (v6.5.1) by https://github.com/jhipster/generator-jhipster/pull/10776 and https://github.com/jhipster/generator-jhipster/pull/10767

Let me know if you have further issues.

**Comment by harald2013** [comment]
Thank you, I can confirm that it is fixed in 6.5.1.
